### PR TITLE
feat(aws_kinesis_firehose_source): Configure multiple access keys

### PR DESCRIPTION
**Pull request by dizlv**
This should address #14106

### CODE REVIEW

File: src/sources/aws_kinesis_firehose/mod.rs [hunk 1]
```diff
@@ -40,6 +40,8 @@ pub struct AwsKinesisFirehoseConfig {
     #[configurable(metadata(docs::examples = "A94A8FE5CCB19BA61C4C08"))]
     access_key: Option<SensitiveString>,
 
+    access_keys: Option<Vec<SensitiveString>>,
```
Comment (jszwedko) [inline src/sources/aws_kinesis_firehose/mod.rs:43]: You'll want to add the `configurable` macro here that you see on the above options as well as a doc comment as this comment will be used to drive the documentation.

File: src/sources/aws_kinesis_firehose/mod.rs [hunk 2]
```diff
@@ -40,6 +40,8 @@ pub struct AwsKinesisFirehoseConfig {
     #[configurable(metadata(docs::examples = "A94A8FE5CCB19BA61C4C08"))]
     access_key: Option<SensitiveString>,
```
Comment (jszwedko) [inline src/sources/aws_kinesis_firehose/mod.rs:41]: We should deprecate this option in favor of the new one. See https://github.com/vectordotdev/vector/blob/master/docs/DEPRECATION.md for the process.

**Review by jszwedko — COMMENTED** [review]
This is looking good @dizlv !

File: src/sources/aws_kinesis_firehose/filters.rs [hunk 3]
```diff
@@ -7,6 +7,7 @@ use snafu::ResultExt;
 use vector_common::internal_event::{BytesReceived, Protocol};
 use vector_core::config::LogNamespace;
 use warp::{http::StatusCode, Filter};
+use vector_common::sensitive_string::SensitiveString;
```
Comment (spencergilbert) [inline src/sources/aws_kinesis_firehose/filters.rs:10]:
```suggestion
use vector_common::{
    internal_event::{BytesReceived, Protocol},
    sensitive_string::SensitiveString,
};
use vector_core::config::LogNamespace;
use warp::{http::StatusCode, Filter};
```

Nit.

**Comment by spencergilbert** [comment]
Just checking @dizlv - this is still WIP and doesn't necessarily need a review until you Ready it?

**Comment by dizlv** [comment]
@spencergilbert I'm currently trying to compile binary to test this feature in my setup, but I do not plan to make any other changes, except code for passing around strings in tests. Probably you want to take a look on it and assist me a little – I had no time to review documentation yet on cloning & referencing in similar codes?

**Comment by dizlv** [comment]
@spencergilbert hey, I have added deprecation notes and about to test this with actual Firehose setup. Other than that and `String` cloning in tests I think I don't have any new changes and it should be ready to move from draft to the PR.

**Comment by dizlv** [comment]
@spencergilbert I believe I've addressed requested changes

File: src/sources/aws_kinesis_firehose/mod.rs [hunk 4]
```diff
@@ -37,9 +37,18 @@ pub struct AwsKinesisFirehoseConfig {
     ///
     /// AWS Kinesis Firehose can be configured to pass along a user-configurable access key with each request. If
     /// configured, `access_key` should be set to the same value. Otherwise, all requests will be allowed.
+    ///
+    /// This option has been deprecated, the `access_keys` option should be used instead.
     #[configurable(metadata(docs::examples = "A94A8FE5CCB19BA61C4C08"))]
```
Comment (spencergilbert) [inline src/sources/aws_kinesis_firehose/mod.rs:42]:
```suggestion
    /// This option has been deprecated, the `access_keys` option should be used instead.
    #[configurable(deprecated)]
    #[configurable(metadata(docs::examples = "A94A8FE5CCB19BA61C4C08"))]
```

Sorry, still nitting 😄

**Review by spencergilbert — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**